### DISCUSSION TITLE
Wait-OVTaskComplete -timeout seems not working

### DISCUSSION THREAD

**Discussion by PetrMa** [HewlettPackard/POSH-HPOneView#585] [Q&A] [answered]
Hello,

I have a PS script which update my HP server with the latest PSP from OneView via command:
Get-OVServerProfile | ?{$_.Name -eq $ShortName} | Update-OVServerProfile -Confirm:$false | Wait-OVTaskComplete -Timeout (New-TimeSpan -Minutes 60)

I applied the parameter -Timeout and set it to 60 minutes because the PSP is not applied in default timeout 20 minutes but it seems that this parameters is ignored because the script cancel the operation after 20 minutes and continues with next lines.

Could you help me what´s wrong please?

Thanks Regards,
Petr

**Comment by ChrisLynchHPE** [comment, marked as answer]
What version of the library are you using?

**Reply by ChrisLynchHPE** [reply]
And looking at the code for `Wait-OVTaskComplete`, the value you are trying to provide to `-Timeout` is supposed to be honored.  So not sure what is happening, unless you have a very old version of the library.  Please provide the output from `$PSLibraryVersion | CLIP`.

**Reply by PetrMa** [reply]
Hello,

we´re using version 6.1. I mean this is the last known version:

Get-OVVersion -ApplianceVer
ApplianceVersion: 6.10.00.437859.00 6.10.2778.2020 C:\Program Files\WindowsPowerShell\Modules\HPEOneView.610\6.10.2778.2020

As I wrote, the command 
Get-OVServerProfile | ?{$_.Name -eq $ShortName} | Update-OVServerProfile -Confirm:$false | Wait-OVTaskComplete -Timeout (New-TimeSpan -Minutes 60)

running exactly 20 minutes (which is the default timeout) and then it stopped and script continues with next lines. 
I also tried the command
Get-OVServer -ServerName $ShortName | Start-OVServer | Wait-OVTaskComplete -Timeout (New-TimeSpan -Minutes 60)
and there the timeout seems to work.
I´am not sure what´s wrong. Is the any way how to check if Update-OVServerProfile command finished or still running?

Let me know if I can provide some debug info.

Thanks, Petr

**Reply by ChrisLynchHPE** [reply]
You need to include the `-Async` parameter with your call to `Update-OVServerProfile`.  Your call should be:

```powershell
Get-OVServerProfile | ?{$_.Name -eq $ShortName} | Update-OVServerProfile -Confirm:$false -Async | Wait-OVTaskComplete -Timeout (New-TimeSpan -Minutes 60)
```

**Reply by PetrMa** [reply]
> You need to include the `-Async` parameter with your call to `Update-OVServerProfile`. Your call should be:
> 
> ```powershell
> Get-OVServerProfile | ?{$_.Name -eq $ShortName} | Update-OVServerProfile -Confirm:$false -Async | Wait-OVTaskComplete -Timeout (New-TimeSpan -Minutes 60)
> ```

Hello Chris,
it seems that -Async command solve my issue and now the script works fine. Thanks, Petr

**Reply by ChrisLynchHPE** [reply]
Thanks for reporting back!  Glad to have solved your issue.
